### PR TITLE
Issues/19

### PR DESCRIPTION
**Pull request by basz**
addresses #19

wip

### CODE REVIEW

**Comment by basz** [comment]
@sandrokeil I found this solution to be very nice (If I may say so myself ;-p). What do you think? One block that simply can be removed when next major release lands. cc @prolic

**Review by prolic — APPROVED** [review]
Looks great! @sandrokeil what to you think?

File: src/Container/PdoSnapshotStoreFactory.php [hunk 1]
```diff
@@ -59,9 +59,25 @@ public static function __callStatic(string $name, array $arguments): PdoSnapshot
...
+            }
+
+            $config = &$config[$this->configId] ?? [];
+
+            if (! isset($config['connection']) && isset($config['connection_service'])) {
+                $config['connection'] = $config['connection_service'];
+            }
+
+            unset($config['connection_service']);
+        })($config);
```
Comment (sandrokeil) [inline src/Container/PdoSnapshotStoreFactory.php:76]: I'm not sure if we should unset the `connection_service` key. Maybe someone is referencing it.

**Comment by sandrokeil** [comment]
Good work :+1:

File: src/Container/PdoSnapshotStoreFactory.php (hunk 1 above)
Comment (prolic) [reply src/Container/PdoSnapshotStoreFactory.php:76]: Agree, @basz can you remove this unset call?

**Comment by prolic** [comment]
@basz can you also check the other snapshot store implementations?

**Comment by basz** [comment]
If true I agree, but how would someone use the old key inside this factory?

**Comment by sandrokeil** [comment]
Not inside the factory. You can retrieve the whole config everywhere in the application where you have access to the service manager. But yes, I guess it's very limited in usage. 

Please add an assertion at the end of the test case, that if you retrieve the config again from the service manager, the key `connection` is also set, because of "value by reference" usage.

**Comment by basz** [comment]
When I add `$config = $container->get('config');` the after the PR'd block `connection_service` reappears . It's not referencing the configuration globally, therefore I don't think this would be an issue. This makes sense to me because an array isn't referenced, only objects in it.

**Comment by prolic** [comment]
@basz good job !! :+1: thanks a lot

**Comment by prolic** [comment]
I'll release all of the snapshot store implementation together, once the change is made everywhere

**Comment by basz** [comment]
done in memcached and prooph/docs. proophessor-do should wait until released

**Comment by sandrokeil** [comment]
@basz That's odd. I've tested it in a project. Maybe it depends if `$config` is an `ArrayObject` or a plain `array`. /cc @prolic 

Added a PR https://github.com/prooph/pdo-snapshot-store/pull/22

I remember that Zend Expressive has used `ArrayObject` in the past for `$config`.

**Comment by basz** [comment]
Yeah, that must be it. One solution would be to clone the $config if it’s an object, but I don’t think it’s worth it. We can simply leave the key (don’t unset). That ensures there are no side effects outside of the factory and that’s the point...

So, you your pr should be accepted

**Pull request merged**